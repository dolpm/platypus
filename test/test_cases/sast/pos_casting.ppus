pipe main [] |> unit {
  int x = -5;
  float y = Float <| [-x+5];
  
  int z = Int <| [y];

  bool t = true;
  int p = Int <| [t];

  char c = 'C';

  float centipede = Float <| [Int <| [c]];

  |> ();
}