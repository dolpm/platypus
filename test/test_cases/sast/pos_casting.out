pipe main [] |> unit {
{(id: main_with_args)
{(id: main_wrapper)
  int x = (int : -(int : 5));
  float y = (float : Float <| [(int : (int : -(int : x)) + (int : 5))]);
  int z = (int : Int <| [(float : y)]);
  bool t = (bool : true);
  int p = (int : Int <| [(bool : t)]);
  char c = (char : 'C');
  float centipede = (float : Float <| [(int : Int <| [(char : c)])]);
  |> (unit : ());
}
}
}
